File: apps/ai-service/app/routes/summarize.py
from fastapi import APIRouter, Depends

from app.models.schemas import SummarizeRequest, SummarizeResult
from app.providers import get_provider
from app.security import require_internal_secret

router = APIRouter()


@router.post("/summarize", response_model=SummarizeResult)
async def summarize(
    request: SummarizeRequest,
    _: None = Depends(require_internal_secret),
) -> SummarizeResult:
    provider = get_provider()
    return await provider.summarize(request)

File: apps/ai-service/src/routes/summarize.ts
import { Router } from "express";

import { getProvider } from "../providers/index.js";
import { requireInternalSecret } from "../security.js";
import { SummarizeRequestSchema } from "../types.js";

export const summarizeRouter = Router();

summarizeRouter.post("/summarize", requireInternalSecret, async (req, res) => {
  const parsed = SummarizeRequestSchema.safeParse(req.body);
  if (!parsed.success) {
    res.status(400).json({ error: "Invalid request", details: parsed.error.flatten() });
    return;
  }

  try {
    const result = await getProvider().summarize(parsed.data);
    res.json(result);
  } catch (err) {
    res.status(502).json({ error: "AI provider failed", detail: String(err) });
  }
});
